FraihmworkHealthAndIntegrityApi.yaml

```diff
@@ -0,0 +1,1361 @@
+        timeout:
+          type: integer
+          minimum: 1
+          example: 100
+          description: Seconds within which a component update is expected from the client. If no update is received, the component will consider the component lifeless.
+        reporter:
+          type: string
+          format: uuid
+          description: The unique identifier of the reporter of this component, if applicable. This assumes that the reporting entity is also a component with a UUID already assigned. This is to allow for components to report status on other components' (such as child entities) behalf.
+        extensions:
+          description: The addition for any component extensions that are optionally included to provide stereotyped information for a particular component. 
+          type: object
+          additionalProperties: true
+
+      required: 
+        - name
+        - state
+
+    Version:
+      type: object
+      description: The version of the component. For maximum flexibility, this is provided as string values to allow for different versioning schemes.
+      properties:
+        hardware:
+          type: string
+          description: The version of the component's hardware package, if applicable
+        software:
+          type: string
+          description: The version of the component's software package, if applicable
+        firmware:
+          type: string
+          description: The version of the component's firmware package, if applicable
+        configuration:
+          type: string
+          description: The version of the component's configuration, if applicable
+
+    ComponentRegistrationInformation:
+      type: object
+      properties:
+        uuid:
+          type: string
+          format: uuid
+        name:
+          type: string
+          description: The name of the component registering the unique identifier
+        description:
+          type: string
+          description: A brief description of the component
+        physicalLocation:
+          type: string
+          description: A physical location describing where the component is
+        networkAddress:
+          type: string
+          description: A network location describing where the component is
+        systemTimeout:
+          type: integer
+          minimum: 1
+          description: Seconds within which a component update is expected from the client. If no update is received, FRAIHMWORK will consider the component lifeless.
+        timeOfRegistration:
+          type: string
+          format: date-time
+          description: The time that the component was first registered. Formatted according to RFC 3339 section 5.6.
+      required:
+        - uuid
+        - name
+
+    State:
+      type: object
+      properties:
+        state:
+          type: string
+          description: "
+          This is a high level description of the component or subsystem state. <br> <br>
```

Comment:
subcomponent. Though, Subcomponents are components, maybe not go into topology in the State description?